### PR TITLE
feat(combobox): add prop to pass classes to the empty list item

### PR DESCRIPTION
**Pull request by josedialpad**
# add prop to pass classes to the empty list item

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

Add a prop to apply classes in the empty list element. In the product [they are using the `dt-default-list-item` class](https://github.com/dialpad/firespotter/pull/24236#discussion_r1024216301) to apply certain styles, with this PR they can use utility classes in the empty list element.

## :pencil: Checklist

- [x] I have reviewed my changes
- [x] I have added tests
- [x] I have added all relevant documentation
- [ ] I have validated components with a screen reader
- [ ] I have validated components keyboard navigation
- [ ] I have considered the performance impact of my change
- [ ] I have checked that my change did not significantly increase bundle size
- [ ] I am exporting any new components or constants in the index.js in the component directory
- [ ] I am exporting any new components or constants in the index.js in the root

### CODE REVIEW

File: components/combobox/combobox_empty-list.vue [hunk 1]
```diff
@@ -4,9 +4,11 @@
...
     <dt-list-item
-      class="d-d-flex d-jc-center"
+      class="dt-empty-list-item d-fs-200 d-lh4 d-py4 d-px12 d-d-flex d-jc-center d-ai-center"
       role="option"
       navigation-type="none"
+      type="custom"
+      :class="itemClass"
     >
       <span>{{ message }}</span>
     </dt-list-item>
```
Comment (braddialpad) [inline components/combobox/combobox_empty-list.vue]: Would it be better to just wrap this list item in a slot an allow them to override it however they wish?

File: components/combobox/combobox_empty-list.vue [hunk 2]
```diff
@@ -4,9 +4,11 @@
     data-qa="dt-combobox-empty-list"
   >
     <dt-list-item
-      class="d-d-flex d-jc-center"
+      class="dt-empty-list-item d-fs-200 d-lh4 d-py4 d-px12 d-d-flex d-jc-center d-ai-center"
```
Comment (braddialpad) [inline components/combobox/combobox_empty-list.vue]: with so many default utility classes in here i'd worry that some of them would not be overridden by `itemclass` due to `!important`

Comment (josedialpad) [reply components/combobox/combobox_empty-list.vue]: yeah I mean if you use other utility classes will override due to the order, but using custom classes you will need to use `!important` in the already defined default classes. So moved the default styles to a class. https://github.com/dialpad/dialtone-vue/pull/601/commits/e65f25683e0256e9fa1a7ad52e723e0042312057

File: components/combobox/combobox_empty-list.vue (hunk 1 above)
Comment (josedialpad) [reply components/combobox/combobox_empty-list.vue]: Yeah to be consistent with other components too, so added a slot to use here https://github.com/dialpad/dialtone-vue/pull/601/commits/ee7edea7b224a308e56286b0dd0b1cc3c0b5677b

**Review by braddialpad — APPROVED** [review]
Looks good!

**Pull request merged**